LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/DefaultLoRaDataRequestHandler.cs

```diff
@@ -64,514 +65,541 @@ public class DefaultLoRaDataRequestHandler : ILoRaDataRequestHandler
 
         public async Task<LoRaDeviceRequestProcessResult> ProcessRequestAsync(LoRaRequest request, LoRaDevice loRaDevice)
         {
-            if (request is null) throw new ArgumentNullException(nameof(request));
-            if (loRaDevice is null) throw new ArgumentNullException(nameof(loRaDevice));
+            ArgumentNullException.ThrowIfNull(request, nameof(request));
+            ArgumentNullException.ThrowIfNull(loRaDevice, nameof(loRaDevice));
 
             List<Task> deferredTasks = null;
+            IAsyncDisposable deviceConnectionActivity = null;
 
-            var timeWatcher = request.GetTimeWatcher();
-
-            var loraPayload = (LoRaPayloadData)request.Payload;
-            this.d2cPayloadSizeHistogram?.Record(loraPayload.Frmpayload.Length);
-
-            var payloadFcnt = loraPayload.Fcnt;
+            try
+            {
+                ExceptionDispatchInfo edi = null;
+                LoRaDeviceRequestProcessResult result = null;
 
-            var payloadFcntAdjusted = LoRaPayloadData.InferUpper32BitsForClientFcnt(payloadFcnt, loRaDevice.FCntUp);
-            this.logger.LogDebug($"converted 16bit FCnt {payloadFcnt} to 32bit FCnt {payloadFcntAdjusted}");
+                try
+                {
+                    result = await ProcessAsync();
+                }
+#pragma warning disable CA1031 // Do not catch general exception types (exception is rethrown)
+                catch (Exception ex)
+#pragma warning restore CA1031 // Do not catch general exception types
+                {
+                    edi = ExceptionDispatchInfo.Capture(ex);
```

Comment:
I am not sure this is needed and unless I am overlooking something, doesn't all of this simply boil down to the following? That is, when the data request processing finishes (successfully or otherwise), we simply log errors in forked tasks (if any).

```c#
List<Task> deferredTasks = null;
IAsyncDisposable deviceConnectionActivity = null;

try
{
    return await ProcessAsync();
}
finally
{
    if (deviceConnectionActivity is { } someDeviceConnectionActivity)
        await someDeviceConnectionActivity.DisposeAsync();

    if (deferredTasks is { } someDeferredTasks)
    {
        _ = await Task.WhenAny(Task.WhenAll(someDeferredTasks));

        var exception = someDeferredTasks.GetExceptions() switch
        {
            { Count: 1 } exs => exs[0],
            { Count: > 1 } exs => new AggregateException(exs),
            _ => null
        };

        if (exception is { } someException)
            this.logger.LogError(someException, "Processing of deferred tasks failed.");
    }
}
```

Reply:
Valid point, I changed the semantics of secondary task failures (these do not fail the message processing) in 8a82a1f3a5182e059c5e29f4cc08c7495d6eb485 and adapted the tests accordingly. Will update the PR description.